extensions/sql/contract-definition/store/src/main/java/org/eclipse/dataspaceconnector/sql/contractdefinition/store/SqlContractDefinitionStore.java

```diff
@@ -0,0 +1,244 @@
+                SqlContractDefinitionTables.CONTRACT_DEFINITION_COLUMN_ACCESS_POLICY,
+                SqlContractDefinitionTables.CONTRACT_DEFINITION_COLUMN_CONTRACT_POLICY,
+                SqlContractDefinitionTables.CONTRACT_DEFINITION_COLUMN_SELECTOR);
+
+
+        transactionContext.execute(() -> {
+            try (Connection connection = dataSource.getConnection()) {
+                executeQuery(connection, query,
+                        definition.getId(),
+                        writeObject(definition.getAccessPolicy()),
+                        writeObject(definition.getContractPolicy()),
+                        writeObject(definition.getSelectorExpression()),
+                        definition.getId());
+            } catch (Exception exception) {
+                throw new EdcPersistenceException(exception);
+            }
+        });
+    }
+
+    @Override
+    public void delete(String id) {
+        Objects.requireNonNull(id);
+
+        String query = String.format("DELETE FROM %s WHERE %s = ?",
+                SqlContractDefinitionTables.CONTRACT_DEFINITION_TABLE,
+                SqlContractDefinitionTables.CONTRACT_DEFINITION_COLUMN_ID);
+
+        transactionContext.execute(() -> {
+            try (Connection connection = dataSource.getConnection()) {
+                executeQuery(connection, query,
+                        id);
+            } catch (Exception exception) {
+                throw new EdcPersistenceException(exception);
+            }
+        });
+    }
+
+    private String writeObject(Object object) {
+        try {
+            String className = object.getClass().getName();
+            String content = OBJECT_MAPPER.writeValueAsString(object);
+
+            Envelope envelope = new Envelope(className, content);
+
+            return OBJECT_MAPPER.writeValueAsString(envelope);
+        } catch (Exception e) {
+            throw new EdcException(e.getMessage(), e);
+        }
+    }
+
+    enum ContractDefinitionMapper implements ResultSetMapper<ContractDefinition> {
+        INSTANCE;
+
+        @Override
+        public ContractDefinition mapResultSet(ResultSet resultSet) throws Exception {
+            return ContractDefinition.Builder.newInstance()
+                    .id(resultSet.getString(SqlContractDefinitionTables.CONTRACT_DEFINITION_COLUMN_ID))
+                    .accessPolicy(readObject(resultSet.getString(SqlContractDefinitionTables.CONTRACT_DEFINITION_COLUMN_ACCESS_POLICY)))
+                    .contractPolicy(readObject(resultSet.getString(SqlContractDefinitionTables.CONTRACT_DEFINITION_COLUMN_CONTRACT_POLICY)))
+                    .selectorExpression(readObject(resultSet.getString(SqlContractDefinitionTables.CONTRACT_DEFINITION_COLUMN_SELECTOR)))
+                    .build();
+        }
+    }
+
+    private static final class Envelope {
+        private String className;
+        private String content;
+
+        public Envelope() {
```

Comment:
not needed. Jackson can use any constructor with `@JsonCreator`. But as I mentioned, I think the `Envelope` class in itself is superfluous.